### ISSUE TITLE
Screen readers cannot update caret offset in code editor

### ISSUE THREAD

**Issue by joanmarie** [microsoft/vscode#98256]
Steps to reproduce:
1. Create a document in the Edtior
2. Type "hello world" and press Return, leaving the caret/cursor on line 2
3. Launch Accerciser, locate the Editor in the accessibility tree (it has role of "entry" and name of "Untitled-1"
4. In Accerciser's iPython console, type the following:

```
text = acc.queryText()
text.setCaretOffset(0)
```

Expected result: The caret would move to the "h" (because that's the character at offset 0)

Actual result: The caret does not move.

Impact: Orca has a SayAll command which allows the user to give a single command / keystroke which causes Orca to speak the full content starting from the present location. When the user reaches the end of the document or interrupts speech by pressing a key like Ctrl, Orca uses ATK/AT-SPI2 to set the caret offset at the last location read. Doing so means the user is where they think they are. ("Orca just read 'foo bar', so I must be on the line with 'foo bar'".) Unfortunately, because `setCaretOffset()` fails to work in the Editor, Orca is unable to make this update and the user remains at the point where the SayAll command was given.

@isidorn: Is this due to the fact that VSCode has a special `textarea` which it uses to make the Editor contents available to screen readers? If so, could you pay attention to the caret being moved in that `textarea` and update the real location within the Editor?

**Comment by isidorn** [comment]
@joanmarie yes this seems to be due to the fact that we are using a `textArea`. So the gist is
Orca updates caret position in the `textArea` and the Editor does not reflect that.

Idealy we would listen on cursor change events in the `textArea` somewhere [here](https://github.com/microsoft/vscode/blob/66ef23d401b24444d1465aa4bd6db369c1632de2/src/vs/editor/browser/controller/textAreaInput.ts#L312) and update the cursor location in the real editor.
Googled a bit and I could not find an event for "caret updated". Though i migtht be missing something.

fyi @alexdima

**Comment by alexdima** [comment]
@isidorn We do listen to selection changes especially for cases where it gets via Braille displays. That was confirmed to be working, although I don't own one of those devices to try it out.

Here it is -- https://github.com/microsoft/vscode/blob/69dadffac58d0993b574b9d04e5823f6367481a1/src/vs/editor/browser/controller/textAreaInput.ts#L382

**Comment by isidorn** [comment]
@alexdima great pointer, thanks a lot!
@joanmarie so I can produce a build with that if statment changed, or if you have time you can just change that line at `textAreaInput:409` and run vscode out of source. Let me know what works for you.

**Comment by joanmarie** [comment]
@isidorn: Removing the check for windows solved it for me. Thanks!

I wonder if VoiceOver might also have a need to reset the caret in that textarea?

**Comment by isidorn** [comment]
@joanmarie great, thanks for trying it out.
I would just remove the check to allow it for all platforms as long as Chrome
Though let's do it start of next week June so we catch potential issues, since we are wrapping up our release now.

**Comment by isidorn** [comment]
I have pushed a change that we allow selection change events to update the selection not only on Win, but on Mac and Linux as well.
Decided to push this start of milestone so we catch potential regressions
@alexdima please let me know if you would like me to test this in a perticular way. Currently I just tried it out with vscode out of source on the mac.

**Comment by isidorn** [comment]
@joanmarie did you get a chance to try this out with vscode insiders?
Asking so we know if this works fine so we add a verified label.

**Comment by jvesouza** [comment]
@isidorn based on the test I did it seems to be correct.

Steps used in the test:

* Launch VSCode and orca.
* Open a file with some content.
* Use the orca command 'say all' to read the file.
* While the orca is reading press the ctrl key to stop reading.
* Use say all again.

as expected, the orca continues to read from where it was stopped.

**Comment by isidorn** [comment]
@jvesouza thanks a lot, really appreciate it! Adding verified label.
